From bdcf762dd8955bdf83cc0078a32fa2e883e42915 Mon Sep 17 00:00:00 2001
From: Limeng <Meng.Li@windriver.com>
Date: Mon, 8 Jun 2020 15:53:59 +0800
Subject: [PATCH 2/9] driver: crypto: caam: set aes_rn variable for imx8
 platform

This issue is introduced by commit e9b8b9945c2("MLK-15473-1:
crypto: caam: Add CAAM driver support for iMX8 soc family").
When get this patch from nxp-imx-sdk4.19, forgot to process the
aes_rn variable for imx8qm platform. imx8 SoC has seco(Security
Controller), so it is need to set aes_rn with Job Ring.

Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/caam/caamalg.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/drivers/crypto/caam/caamalg.c b/drivers/crypto/caam/caamalg.c
index 0319e02fac61..4950d0152ad4 100644
--- a/drivers/crypto/caam/caamalg.c
+++ b/drivers/crypto/caam/caamalg.c
@@ -3602,8 +3602,13 @@ static int __init caam_algapi_init(void)
 		ccha_inst = 0;
 		ptha_inst = 0;
 
-		aes_rn = rd_reg32(&priv->ctrl->perfmon.cha_rev_ls) &
-			 CHA_ID_LS_AES_MASK;
+		if (priv->has_seco) {
+			aes_rn = rd_reg32(&priv->jr[0]->perfmon.cha_rev_ls) &
+				 CHA_ID_LS_AES_MASK;
+		} else {
+			aes_rn = rd_reg32(&priv->ctrl->perfmon.cha_rev_ls) &
+				 CHA_ID_LS_AES_MASK;
+		}
 		gcm_support = !(aes_vid == CHA_VER_VID_AES_LP && aes_rn < 8);
 	} else {
 		u32 aesa, mdha;
-- 
2.17.1

